DuckDuckGo/PrivacyProtectionController.swift

```diff
@@ -69,69 +55,53 @@ class PrivacyProtectionController: UIViewController {
             showInitialScreen()
         }
 
-        applyTheme(ThemeManager.shared.currentTheme)
     }
 
     private func showError(withText errorText: String) {
         guard let controller = storyboard?.instantiateViewController(withIdentifier: "Error") as? PrivacyProtectionErrorController else { return }
         controller.errorText = errorText
-        embeddedController.pushViewController(controller, animated: true)
+        pushViewController(controller, animated: true)
     }
 
     private func showBlockerListError() {
         guard let controller = storyboard?.instantiateViewController(withIdentifier: "Error") as? PrivacyProtectionErrorController else { return }
         controller.errorText = UserText.privacyProtectionReloadBlockerLists
         controller.delegate = self
-        embeddedController.pushViewController(controller, animated: true)
+        pushViewController(controller, animated: true)
     }
 
     private func showInitialScreen() {
         guard let controller = storyboard?.instantiateViewController(withIdentifier: "InitialScreen")
             as? PrivacyProtectionOverviewController else { return }
-        embeddedController.pushViewController(controller, animated: true)
+        pushViewController(controller, animated: true)
         updateViewControllers()
     }
 
-    private func initOmniBar() {
-        omniBar = OmniBar.loadFromXib()
-        omniBar.frame = omniBarContainer.bounds
-        omniBarContainer.addSubview(omniBar)
-        omniBar.textField.text = omniBarText
-        omniBar.updateSiteRating(siteRating, with: storageCache)
-        omniBar.startBrowsing()
-        omniBar.omniDelegate = self
-        omniBar.textField.addTarget(self, action: #selector(onTextFieldTapped), for: .touchDown)
-    }
-
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if let controller = segue.destination as? UINavigationController {
-            embeddedController = controller
-            updateViewControllers()
-        }
-    }
-
-    @objc func onTextFieldTapped() {
-        dismiss(animated: true) {
-            self.delegate?.omniBarTextTapped()
-        }
+    override func pushViewController(_ viewController: UIViewController, animated: Bool) {
+        viewController.title = siteRating?.domain
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(done))
+        viewController.navigationItem.leftBarButtonItem = doneButton
+        super.pushViewController(viewController, animated: animated)
```

Comment:
To be honest Privacy Dashboard navigation seems (even more) confusing now... In a place where I'd expect back button to be, there is 'Done'.

Since we made it work (more or less) like a regular modal now, why not just re-introduce back button for the navigation and have a 'Done' button on the right?

Reply:
This was a UX request - let's discuss it on Asana.

Reply:
I'm going to leave this one for now given Robert's comments and I think it'll make more sense in iOS 13 when that happens.